bmwqemu.pm

```diff
@@ -260,15 +260,16 @@ sub current_test {
 sub update_line_number {
     return unless current_test;
     return unless current_test->{script};
-    my $out    = "";
-    my $ending = quotemeta(current_test->{script});
-    for my $i (1 .. 10) {
+    my @out;
+    my $ending  = quotemeta(current_test->{script});
+    my $casedir = $vars{CASEDIR} // '';
+    for (my $i = 10; $i > 0; $i--) {
         my ($package, $filename, $line, $subroutine, $hasargs, $wantarray, $evaltext, $is_require, $hints, $bitmask, $hinthash) = caller($i);
-        last unless $filename;
-        next unless $filename =~ m/$ending$/;
-        $logger->debug("$filename:$line called $subroutine");
-        last;
+        next unless $filename && $filename =~ /$casedir/;
```

Comment:
`$filename =~ /$casedir/;` - isn't it will always return true ? when it will be false ?

Reply:
`$casedir` should be quoted here I think (might contain special regex chars):
`$filename =~ /\Q$casedir/`

Reply:
`$filename =~ /$casedir/` returns false for any function which is not defined within the test distribution, e.g. any system installed functions which we want to ignore. In before we would *only* match on the current test module. My idea was to log any function within the test distribution, but not system functions.

Reply:
can you give me example of system function which I currently can see in log ?

Reply:
You can't because currently we log *only* for test modules. This is extended now to output multiple locations but exclude the ones outside the scope of casedir

Reply:
added the `\Q`